packages/nextra/src/compile.ts

```diff
@@ -77,45 +81,50 @@ export async function compileMdx(
   const mdxOptions = {
     ...(loaderOptions.mdxOptions || {}),
     // You can override MDX options in the frontMatter too.
-    ...frontMatter.mdxOptions
+    ...frontMatter.mdxOptions as Record<any, unknown>
   }
+  const format = mdxOptions.format || 'mdx'
+  const compiler =
+    (useCachedCompiler && cachedCompilerForFormat[format]) ||
+    (cachedCompilerForFormat[format] = createCompiler({
+      jsx: mdxOptions.jsx || false,
+      outputFormat: mdxOptions.outputFormat || 'function-body',
+      providerImportSource: 'nextra/mdx',
+      format,
+      // https://github.com/hashicorp/next-mdx-remote/issues/307#issuecomment-1363415249
+      development: false,
+      remarkPlugins: [
+        ...(mdxOptions.remarkPlugins || []),
+        remarkGfm,
+        remarkHeadings,
+        loaderOptions.staticImage && remarkStaticImage,
+        loaderOptions.flexsearch &&
+          structurize(structurizedData, loaderOptions.flexsearch),
+        loaderOptions.readingTime && readingTime,
+        loaderOptions.latex && remarkMath
+      ].filter(truthy),
+      rehypePlugins: [
+        ...(mdxOptions.rehypePlugins || []),
+        parseMeta,
+        loaderOptions.codeHighlight &&
+          ([
+            rehypePrettyCode,
+            {
+              ...rehypePrettyCodeOptions,
+              ...mdxOptions.rehypePrettyCodeOptions
+            }
+          ] as any),
+        [
+          attachMeta,
+          { defaultShowCopyCode: loaderOptions.defaultShowCopyCode }
+        ],
+        loaderOptions.latex && rehypeKatex
+      ].filter(truthy)
+    }))
 
-  const compiler = createCompiler({
-    jsx: mdxOptions.jsx || false,
-    outputFormat: mdxOptions.outputFormat || 'function-body',
-    providerImportSource: 'nextra/mdx',
-    format: mdxOptions.format || 'mdx',
-    // https://github.com/hashicorp/next-mdx-remote/issues/307#issuecomment-1363415249
-    development: false,
-    remarkPlugins: [
-      ...(mdxOptions.remarkPlugins || []),
-      remarkGfm,
-      remarkHeadings,
-      loaderOptions.staticImage && ([remarkStaticImage, { filePath }] as any),
```

Comment:
since filePath will be cached, I removed this parameter from `remarkStaticImage `